[lld-macho] Enable Linker Optimization Hints pass for arm64_32

BertalanD opened the pull request:
The backend emits `.loh` directives for arm64_32 as well. Our pass already handles 32-bit pointer loads correctly (there was an extraneous sanity check for 8-byte pointer sizes, I removed that here), so we can enable them for both arm64 subtargets, as well as our upcoming arm64e support.

This commit moves the functionality out of the arm64 `TargetInfo` into a standalone file.

ellishg commented:
Should we move this code to its own file? Maybe `ARM64LOH.cpp`?

BertalanD commented:
Good idea, I factored the functionality out of `TargetInfo`.

nico reviewed (approved):
Thanks!

I like putting no-op code moves and behavior changes into separate commits, so personally I'd land this as one commit that moves some code to a new file (without changing any behavior) and a second commit that changes behavior.

But up to you :)

BertalanD commented:
Committed NFC code move as fb3972dd06cbc1b0a5305f81b0c2d74e44dbea41

BertalanD committed:
[lld-macho] Enable Linker Optimization Hints pass for arm64_32

The backend emits `.loh` directives for arm64_32 as well. Our pass
already handles 32-bit pointer loads correctly (there was an extraneous
sanity check for 8-byte pointer sizes, I removed that here), so we can
enable them for all arm64 subtargets, including our upcoming arm64e
support.
